notebooks/sbgn.ipynb

```diff
@@ -30,12 +39,53 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import libsbgnpy.libsbgn as libsbgn\n",
+    "import os\n",
+    "\n",
     "from IPython.display import Image\n",
+    "import libsbgnpy.libsbgn as libsbgn\n",
     "from libsbgnpy.libsbgnTypes import ArcClass, GlyphClass, Language\n",
-    "from combine_notebooks import RESULTS_DIR\n",
     "from libsbgnpy import render\n",
-    "import os"
+    "import copy\n",
+    "\n",
+    "from combine_notebooks import RESULTS_DIR"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_sbgn_string(doc_before : libsbgn.sbgn):\n",
+    "    # Creates temporary sbgn file to extract xml as a string\n",
+    "    sbgn_path = str(RESULTS_DIR) + \"/temp.sbgn\"\n",
+    "    os.makedirs(os.path.dirname(sbgn_path), exist_ok=True)\n",
+    "    f_out = sbgn_path\n",
+    "    doc_before.write_file(f_out)\n",
+    "    xml = open(f_out).read()\n",
+    "    os.remove(sbgn_path) \n",
+    "    return xml\n",
+    "\n",
+    "def pretty_print(sbgn_before : libsbgn.sbgn, sbgn_after : libsbgn.sbgn = None):\n",
```

Comment:
I could be utility, but each markdown will need its own implementation because each standard's python library has its own nuances in how the xml needs to be printed.

Let me know if you'd prefer the pretty_print functions be in a utils file.

Reply:
no if it has to be like that that's fine